sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -60,18 +60,24 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
         ctx.INPUT_ROW,
         children.zipWithIndex.map { case (e, i) =>
           val eval = e.genCode(ctx)
-          eval.code + s"""
-            if (${eval.isNull}) {
```

Comment:
it's not easy, for example:
```
boolean isNull = false || false;
if (isNull) ...
```
this should be optimized, but is very hard for our string based codegen framework.
I think it makes more sense to leave them for javac or JIT, again, the readability of generated java code doesn't matter.

BTW, it's a good idea to avoid `zeroOutNullBytes`, if we can figure out a simple and clear way to do it.